deploy/helm-chart/templates/deployment.yaml

```diff
@@ -36,11 +36,13 @@ spec:
         imagePullPolicy: {{ .Values.nginxGateway.image.pullPolicy }}
         name: nginx-gateway
         securityContext:
+          allowPrivilegeEscalation: false
           capabilities:
             add:
             - KILL
             drop:
             - ALL
+          readOnlyRootFilesystem: true
```

Comment:
what about this setting for the data plane container? would it also make sense to do that?

(It should be possible in theory -- we were able to do that in nginmesh project -- just put all the things where we need to write or NGINX needs to write into volumes. One of those things are temp files NGINX can create when proxying https://github.com/nginxinc/nginmesh/blob/584da6de5dbf5b643e53fe43a593bf46f084d670/istio/agent/nginx/template.go#L200-L204 )

Reply:
Yeah, saw this error: `mkdir() "/var/cache/nginx/client_temp" failed (30: Read-only file system)`. Using an emptyDir volume for that path (or I need a new path?) will allow it to be writable?

Reply:
>  Using an emptyDir volume for that path (or I need a new path?) will allow it to be writable?

yep